scipy/spatial/transform/_rotation.pyx

```diff
@@ -2806,54 +2833,237 @@ cdef class Rotation:
-            weights = np.ones(len(b))
+            weights = np.ones(N)
         else:
-            weights = np.asarray(weights)
-            if weights.ndim != 1:
-                raise ValueError("Expected `weights` to be 1 dimensional, got "
-                                 "shape {}.".format(weights.shape))
-            if weights.shape[0] != b.shape[0]:
-                raise ValueError("Expected `weights` to have number of values "
-                                 "equal to number of input vectors, got "
-                                 "{} values and {} vectors.".format(
-                                    weights.shape[0], b.shape[0]))
-            if (weights < 0).any():
-                raise ValueError("`weights` may not contain negative values")
-
+            weights = np.array(weights, dtype=float)
+
+        if weights.ndim != 1:
+            raise ValueError("Expected `weights` to be 1 dimensional, got "
+                             "shape {}.".format(weights.shape))
+        if N > 1 and (weights.shape[0] != N):
+            raise ValueError("Expected `weights` to have number of values "
+                             "equal to number of input vectors, got "
+                             "{} values and {} vectors.".format(
+                                 weights.shape[0], N))
+        if (weights < 0).any():
+            raise ValueError("`weights` may not contain negative values")
+
+        # For the special case of a single vector pair, we use the infinite
+        # weight code path
+        if N == 1:
+            weight_is_inf = np.array([True])
+        else:
+            weight_is_inf = np.isposinf(weights)
+        n_inf = np.sum(weight_is_inf)
+
+        # Check for an infinite weight, which indicates that the corresponding
+        # vector pair is the primary unmoving reference to which we align the
+        # other secondary vectors
+        if n_inf > 1:
+            raise ValueError("Only one infinite weight is allowed")
+
+        elif n_inf == 1:
+            if return_sensitivity:
+                raise ValueError("Cannot return sensitivity matrix with an "
+                                 "infinite weight or one vector pair")
+            a_pri = a[weight_is_inf]
+            b_pri = b[weight_is_inf]
+            a_pri /= _norm3(a_pri[0])
+            b_pri /= _norm3(b_pri[0])
+
+            # We first find the minimum angle rotation between the primary
+            # vectors.
+            cross = np.cross(b_pri[0], a_pri[0])
+            theta = atan2(_norm3(cross), np.dot(a_pri[0], b_pri[0]))
+            if theta < 1e-3:  # small angle Taylor series approximation
+                theta2 = theta * theta
+                r = cross * (1 + theta2 / 6 + theta2 * theta2 * 7 / 360)
+            else:
+                r = cross * theta / np.sin(theta)
+            R_pri = cls.from_rotvec(r)
+
+            if N == 1:
+                # While the else clause below will work for a single pair of
```

Comment:
I think this shortcut is totally intuitive and perhaps it would be strange for someone if we do otherwise. 

So maybe this comment is not necessary.

Reply:
Cut this down